pkg/resources/marketplace/implicit_catalog_source_reconciler.go

```diff
@@ -48,6 +51,37 @@ func (r *ImplicitCatalogSourceReconciler) Reconcile(ctx context.Context) (reconc
 		return reconcile.Result{}, errors.New("catalog source not found for implicit product installation type")
 	}
 
+	r.Log.Infof("Found: ", l.Fields{"catalogSource": catalogSource.Name, "ns": catalogSource.Namespace})
+
+	watchNS, err := k8sresources.GetWatchNamespace()
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "could not get watch namespace in ImplicitCatalogSourceReconciler")
+	}
+	namespaceSegments := strings.Split(watchNS, "-")
+	namespacePrefix := strings.Join(namespaceSegments[0:2], "-") + "-"
+	ns := namespacePrefix + "operator"
+
+	// If the Catalog Source is in the rhoam operator ns but the subscription is for 3Scale then
+	// we need to copy the CS to the 3Scale operator namespace in order for it to be discovered.
+	if catalogSource.Namespace == ns && strings.Contains(subName, "3scale") {
+		r.Log.Info("Copying 3scale Catalog source")
+
+		cs := &coreosv1alpha1.CatalogSource{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      catalogSource.Name,
+				Namespace: namespacePrefix + "3scale-operator",
+			},
+		}
+
+		controllerutil.CreateOrUpdate(ctx, r.Client, cs, func() error {
+			cs.Spec = catalogSource.DeepCopy().Spec
+			return nil
+		})
```

Comment:
Hm, out of curiosity does this mean that we would have two identical CatalogSources? :thinking:  And won't the one in the `redhat-rhoam-3scale-operator` namespace be updated on each reconcile potentially rewriting any changes there might be to the copy?

Reply:
We will have 2 but I don't think it will be overridden on each reconcile as we don't write the CS I believe. Unless you've seen otherwise @makslion? This is a temp solution until Cpaas guys implement a fix.